atomic_reactor/util.py

```diff
@@ -808,11 +809,55 @@ def query_registry(registry_session, image, digest=None, version='v1', is_blob=F
     logger.debug("query_registry: querying {}, headers: {}".format(url, headers))
 
     response = registry_session.get(url, headers=headers)
+    for r in chain(reversed(response.history), [response]):
+        logger.debug("query_registry: [%s] %s", r.status_code, r.url)
+
+    logger.debug("query_registry: response headers: %s", response.headers)
     response.raise_for_status()
 
     return response
 
 
+def guess_manifest_media_type(content):
+    encoding = guess_json_utf(content)
+    try:
+        manifest = json.loads(content.decode(encoding))
+        return manifest['mediaType']
```

Comment:
maybe move this to it's own 'try' ? just to be sure that decode won't raise any KeyError

Reply:
Line 830 catches any KeyError from this.

Reply:
yes, but if decode raises KeyError it should different than KeyError raised from "return manifest['mediaType']"

Reply:
Ah, bytes.decode() can raise a UnicodeDecodeError exception, yes, good point. Changed.